extensions/network/network.impl/src/main/java/org/mqnaas/network/impl/NetworkRootResourceProvider.java

```diff
@@ -82,10 +82,13 @@ public void setRootResources(Collection<IRootResource> rootResources) {
 
 			Specification specification = resource.getDescriptor().getSpecification();
 
+			// Unspecified model (=null) matches all models
+			// Unspecified version (=null) matches all versions
+			// Specified fields (type, model and version !=null) match only the same value
 			boolean matches = true;
-			matches &= (type != null && specification.getType() != null) ? specification.getType().equals(type) : true;
-			matches &= (model != null && specification.getModel() != null) ? specification.getModel().equals(model) : true;
```

Comment:
This implementation was not coherent with the rest.

In case specification.getModel()== null the implementation told they matched.